### ISSUE TITLE
New LabelledButton component

### ISSUE THREAD

**Issue by braaar** [indivorg/kit#11]
Requirements for the LabelledButton component we want:
- Must support various variants (colours, border shapes, border thicknesses)
- Must align icons with text in a proper manner (middle of the icon should align with the middle of the text?)
- Must support icons of different kinds.

Good:
![Screenshot 2021-09-07 at 10 01 30](https://user-images.githubusercontent.com/5765650/132299131-44a919c9-3ba1-42b2-937b-6eebf872a5a2.png)

Bad:
![Screenshot 2021-09-07 at 10 02 00](https://user-images.githubusercontent.com/5765650/132299193-ce6358c9-6050-443f-a2de-ee5db749cc07.png)

**Comment by braaar** [comment]
New style:
<img width="742" alt="Screenshot 2021-09-09 at 10 06 04" src="https://user-images.githubusercontent.com/5765650/132638943-fed78669-b6e7-4e16-8823-1dff549b2f8f.png">

**Comment by cobraz** [comment]
Can this just be a styled `Button`? or maybe `IconButton`? I don't understand the name `LabelledButton`.

**Comment by braaar** [comment]
Very true. I like IconButton. LabelledButton is basically a tautology

**Comment by braaar** [comment]
I'm not sure if we will be able to pull this off with just styling on the button object. We probably need a flex to align the icons properly

**Comment by cobraz** [comment]
Right, let's implement a `<IconButton />` where one of the inputs is an icon (ReactElement).

E.g.

```jsx
<IconButton icon={<Icon />} />
```

Not sure about the name `IconButton`, considering this: https://theme-ui.com/components/icon-button/

**Comment by braaar** [comment]
> Right, let's implement a `<IconButton />` where one of the inputs is an icon (ReactElement).
> 
> E.g.
> 
> ```js
> <IconButton icon={<Icon />} />
> ```
> 
> Not sure about the name `IconButton`, considering this: https://theme-ui.com/components/icon-button/

I like this syntax! I think it will end up being

```ts
import {Icon} from 'react-icons';

<IconButton icon={Icon} />
```

**Comment by braaar** [comment]
`ButtonWithReactIcon`
`ReactIconButton`
`SymbolButton`
`ActionButton`

**Comment by seacurrent** [comment]
ActionButton gets my vote.

**Comment by cobraz** [comment]
I don't know, isn't all buttons meant to have an _action_? I'll do some research, see if find some other examples/names.

**Comment by braaar** [comment]
`SymbolButton`
`ButtonWithSymbol`

**Comment by cobraz** [comment]
How about `AffixButton`? As in a button with `prefix` and `suffix`, I imagine this contract:

```jsx
<AffixButton suffix={<Icon />} />
<AffixButton prefix={<Icon />} />
<AffixButton prefix={<PreIcon />} suffix={<AfterIcon />} />
```

What do you think?

**Comment by braaar** [comment]
The name has its logic, I guess, but it's not intuitive at a glance, in my opinion. Something referring to the fact that the button has an icon, graphic, visual element on it is more intuitive.

I noticed a bug in the current implementation. There is a small margin on the text label which does not trigger the onClick action and also does not show the mouse as a cursor.

https://user-images.githubusercontent.com/5765650/134669788-697cc2e3-8346-4608-8f75-01bb2e0d3627.mov

**Comment by braaar** [comment]
Actually, the entire text area does not trigger the onClick.
